aidial_adapter_openai/utils/chat_completion_response.py

```diff
@@ -0,0 +1,51 @@
+from typing import Any, Iterable, Literal, Self
+
+from aidial_sdk.utils.merge_chunks import merge_chat_completion_chunks
+from pydantic import BaseModel
+
+
+class ChatCompletionResponse(BaseModel):
+    message_key: Literal["delta", "message"]
+    resp: dict = {}
+
+    @property
+    def usage(self) -> Any | None:
+        return self.resp.get("usage")
+
+    @property
+    def is_empty(self) -> bool:
+        return self.resp == {}
+
+    @property
+    def finish_reasons(self) -> Iterable[Any]:
```

Comment:
I think that after using Generic and TypedDict from openai library, it should infer, that 
`finish_reasons -> Iterable[Literal["stop", "length", "tool_calls", "content_filter", "function_call"]]`,
and we can remove this `-> Iterable[Any]:`